testflinger/__init__.py

```diff
@@ -43,9 +44,7 @@ class DefaultConfig:
 
     REDIS_HOST = "localhost"
     REDIS_PORT = "6379"
-    DATA_PATH = os.path.join(
-        os.path.dirname(os.path.abspath(__file__)), "data"
-    )
+    DATA_PATH = Path(__file__).resolve().parent / "data"
```

Comment:
Did you consider using PurePath here? Any advantage to this over using something like this?
DATA_PATH = PurePath(__file__).parent / "data"

Reply:
As I understand, if we use PurePath, we need to specify full path, because [PurePath doesn't access the filesystem](https://docs.python.org/3/library/pathlib.html#pure-paths). But with 	
Path.resolve() we can get the absolute path ([see the first row of the table here](https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module)). That's why I have decided to use Path.
Maybe I'm not right, please, tell me if I miss something

Reply:
resolve() does need to access the real filesystem to check symlinks, but it will only raise an exception (which is not handled here) if you set strict=True.  I'm just wondering if we need to resolve symlinks, or if having DATA_PATH potentially contain a symlink would create any problem with the way we use it. iirc, we are only reading/writing some data files with that dir as a base. We also currently use os.path.abspath, which also doesn't really inspect the filesystem.